internal/git/cli.go

```diff
@@ -462,7 +510,11 @@ func (g *GitCLI) Fetch(refSpec []string, options FetchOptions) error {
 		}
 	}
 
-	args = append(args, "origin")
+	if options.LocalRepository != "" {
+		args = append(args, options.LocalRepository)
```

Comment:
We may need to run two fetches if `tags || options.FetchDepth > 0`... or we may need to tell the merge helper to also fetch tags and/or ensure the minimum fetch depth we will be requesting

Reply:
Addressed for `fetchDepth`, tags currently not supported by the `checkout` action.

Reply:
Tags are fetched implicitly when depth is 0

Reply:
So i just tested the `v1` version of the checkout action with `fetch-depth` set to 0 and it failed, see [Run](https://saas-preprod.beescloud.com/orgs/cloudbees-preprod~cbp/components/2744c987-bc78-4d3c-8473-dd5506cfe2e8/runs/branches/fbf6cdad-3ab8-46a6-87ec-6dbce7e1758c/workflows/6d54cbc2-3062-4894-a5c1-691bab888d9f/runs/81416bd6-dd40-4492-b0b4-f4e2af08f4cc/attempts/1/jobs?job=build&step=s002-0-0) and workflow [my-automation](https://bitbucket.org/cb-pod2/bbprod/src/d7d74e963b298377944550c02699302850123e1e/.cloudbees/workflows/my-automation.yaml#lines-13:17), this is without any of my changes. Looks like a separate bug.